src/Compilers/CSharp/Portable/SymbolDisplay/ObjectDisplay.cs

```diff
@@ -262,7 +262,43 @@ public static string FormatLiteral(string value, ObjectDisplayOptions options)
                 throw new ArgumentNullException("value");
             }
 
-            return FormatString(value, options.IncludesOption(ObjectDisplayOptions.UseQuotes) ? '"' : '\0', escapeNonPrintable: true);
+            var quote = options.IncludesOption(ObjectDisplayOptions.UseQuotes) ? '"' : '\0';
+            PooledStringBuilder pooledBuilder = null;
+            StringBuilder builder = null;
+            if (quote != 0)
+            {
+                pooledBuilder = PooledStringBuilder.GetInstance();
+                builder = pooledBuilder.Builder;
+                builder.Append(quote);
+            }
+            for (int i = 0; i < value.Length; i++)
+            {
+                FormatChar(ref pooledBuilder, value, i, value[i], quote, useLanguageSpecificEscapes: true, useUnicodeEscapes: true);
+            }
+            if (quote != 0)
+            {
+                builder.Append(quote);
+            }
+            return (pooledBuilder == null) ? value : pooledBuilder.ToStringAndFree();
```

Comment:
This appears to be the reason for passing the PooledStringBuilder by reference.  I'm not sure this is worth the extra complexity of keeping track of callee-allocated objects.